docs(kic): use generated CRDs reference (#4813)
Kong/docs.konghq.com: app/_src/kubernetes-ingress-controller/guides/using-kongplugin-resource.md

`KongPlugin` Custom Resource to control proxied requests, including
restricting paths and transforming requests.

> Please see the [concept](/kubernetes-ingress-controller/{{page.kong_version}}/concepts/custom-resources/#KongPlugin)

Reviewer: ```suggestion
See the [concept](/kubernetes-ingress-controller/{{page.kong_version}}/concepts/custom-resources/#KongPlugin)
```

> document for understanding the purpose of `KongPlugin` resource.

Reviewer: ```suggestion
documentation for more information about the purpose of the `KongPlugin` resource.
```

{:.important}

> > The controller does not validate configuration by default. Although not required for this guide, production instances

Reviewer: ```suggestion
> The controller doesn't validate configuration by default. Although not required for this guide, production instances
```

Here, we have successfully set up a plugin which is executed only when a
request matches a specific `Ingress` rule.

## Storing plugin configuration in a Secret

> You can store plugin configuration in a Secret to secure sensitive configuration. To do so, first create a Secret

Reviewer: ```suggestion
You can store plugin configuration in a Secret to secure the sensitive configuration. To do so, first create a Secret
```

> with a key for each field you wish to configure:

Reviewer: ```suggestion
with a key for each field you want to configure:
```

```sh
echo '
apiVersion: v1
kind: Secret
metadata:
  name: plugin-conf-secret
stringData:
  add-response-header: |
    add:
      headers:
        - "demo: injected-by-kong"
type: Opaque
' | kubectl apply -f -
```

> The output is similar to the following:

Reviewer: ```suggestion
Example output:
```

```
secret/plugin-conf-secret created
```

> Then, create a KongPlugin with a `configFrom` field referring that Secret:

Reviewer: ```suggestion
Then, create a KongPlugin with a `configFrom` field referring to that Secret:
```

```sh
echo '
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: add-response-header
configFrom:
  secretKeyRef:
    name: plugin-conf-secret
    key: add-response-header
plugin: response-transformer
' | kubectl apply -f -
```

> The output is similar to the following:

Reviewer: ```suggestion
Example output:
```
